src/filters/oidc/jwks_resolver.cc

```diff
@@ -37,8 +42,22 @@ void DynamicJwksResolverImpl::JwksFetcher::request(
       }
     }
 
+    constexpr auto default_interval_with_no_jwks = std::chrono::seconds(3);
+
+    if (next_schedule_interval > default_interval_with_no_jwks &&
+        parent_->jwks() == nullptr) {
+      // On Kubernetes, depending on the timing of the Pod startup,
```

Comment:
So if I understand correctly, the following error case might still happen:

1. pod comes up.
2. k8s network glitches, jwks request fails for whatever resaon ( could also be due to IdP is temporarily not available)
3. External Authz Check request comes in. However at this point we still don't have jwks ready yet for JWT verification.

Is that right? Ideally we should make the jwks fetching completion as the criteria for the authservice pod [readinessProbe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes)

if above is correct, can we create a TODO and log a bug to track this? Don't have to be done in this PR for sure.

Reply:
I think that it can be better to have the feature to configure `initialDelaySeconds` that acts to start the initial request after these seconds. I will change the implementation of this PR.

Also, The idea seems great. My idea is to have `/health` endpoint that returns 200 not only authservice is fine but JWKs are ready.

Reply:
I think given you have the /health endpoint, and prober logic will prevent us from resulting into the above failure case, right? I would rather keep it that way, since how long to delay, (3seconds), whether to delay is really up to the environmental issue.

A reconciling model of probing /health endpoint would ensure us to eventual get there with less knobs. Customers would also not have to configure one more field. Trying to understand the impact of different values to the startup success.